### PR TITLE
Feature flags: Add a way to subscribe to feature flag changes

### PR DESCRIPTION
**Pull request by philipp-spiess**
Adds a new method to the feature flag provider in order to subscribe to changes for a specific prefix and subscribes the autocomplete codebase to all `cody.autocomplete` changes. When triggered, it will re-initialize just like when settings are changed.

The main motivation is that we want to be able to change experiment rollouts without having to wait for days until all users have reloaded the VS Code extension.

## Test plan

- Added unit tests
- https://github.com/sourcegraph/cody/assets/458591/b32636c8-66be-483e-827b-0d749bab7fc1

### CODE REVIEW

**Comment by philipp-spiess** [comment]
This has been much more involved than I initially thought. There's still some kind of race condition in there that I don't fully understand. I think I'll refactor the feature flag provider to not mix "undefined" feature flags with the preloaded feature flag object but instead cache them in a different object. 😓  cc @valerybugakov if you have any input

**Comment by valerybugakov** [comment]
> not mix "undefined" feature flags with the preloaded feature flag object but instead cache them in a different object

Yes, please! Based on our recent discussion this part is tricky to reason about. Making it more transparent and robust would be great.

**Comment by philipp-spiess** [comment]
@valerybugakov This is ready for review now, I addressed the point I mentioned above

File: vscode/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,15 @@ This is a log of all notable changes to Cody for VS Code. [Unreleased] changes a
 
 ### Added
 
+- Chat: Display Cody icon in the editor title of the chat panels when `cody.editorTitleCommandIcon` is enabled. [pull/2937](https://github.com/sourcegraph/cody/pull/2937)
```
Comment (umpox) [inline vscode/CHANGELOG.md:9]: Looks like a bad merge, needs updating!

**Review by umpox — APPROVED** [review]
Looks good! and v helpful for future experiments. Thanks for adding the tests

File: vscode/CHANGELOG.md (hunk 1 above)
Comment (philipp-spiess) [reply vscode/CHANGELOG.md:9]: ouch, ty!

**Pull request merged**